<a href="https://colab.research.google.com/github/DDD86/Python_new/blob/main/Task_4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import numpy as np
from scipy import stats

In [2]:
# Данные для группы A и B
mean_A = 360
std_A = 40
n_A = 9802

mean_B = 352
std_B = 58
n_B = 9789

In [5]:
# Вычисление стандартной ошибки
se_A = std_A / np.sqrt(n_A)
se_B = std_B / np.sqrt(n_B)

# Вычисление t-статистики
t_stat = (mean_A - mean_B) / np.sqrt(se_A**2 + se_B**2)

# Степени свободы (приближенные, поскольку выборки почти равны)
df = min(n_A - 1, n_B - 1)

# Вычисление p-value для двустороннего теста
p_value = 2 * (1 - stats.t.cdf(np.abs(t_stat), df))

t_stat, p_value

(11.236630785707332, 0.0)

**t-статистика:** 11.24
**p-value:** 0

Это значит, что результат теста очень значимый, даже при уровне доверия 80% (где α = 0.20). Поскольку p-value намного меньше этого уровня, мы можем сказать, что среднее время, которое пользователи проводят, отличается между группами A и B.

**Версия A (средняя 360) лучше, чем версия B (средняя 352). Поскольку разница значима, мы рекомендуем выпустить на продакшен версию A.**